pyiso/eu.py

```diff
@@ -0,0 +1,252 @@
+        if 'UNKNOWN_EXCEPTION' in r.text:
+            self.logger.warn('UNKNOWN EXCEPTION')
+            return False
+        return r.text
+
+    def construct_payload(self):
+        format_str = '%d.%m.%Y'
+        date_str = self.options['start_at'].strftime(format_str) + ' 00:00|UTC|DAY'
+        (TSO_ID, ) = [
+            i['ENTSOe_ID'] for i in control_areas if i['Code'] == self.options['control_area']]
+
+        payload = {
+            'name': '',
+            'defaultValue': 'false',
+            'viewType': 'TABLE',
+            'areaType': 'CTA',
+            'atch': 'false',
+            'dateTime.dateTime': date_str,
+            'biddingZone.values': TSO_ID,
+            'dateTime.timezone': 'UTC',
+            'dateTime.timezone_input': 'UTC',
+            'exportType': 'CSV',
+            'dataItem': 'ALL',
+            'timeRange': 'DEFAULT',
+        }
+        return payload
+
+    def parse_load_response(self, response):
+        df = pd.read_csv(StringIO(response))
+
+        # get START_TIME_UTC as tz-aware datetime
+        df['START_TIME_UTC'], df['END_TIME_UTC'] = zip(
+            *df['Time (UTC)'].apply(lambda x: x.split('-')))
+
+        # Why do these methods only work on Index and not Series?
+        df.set_index(df.START_TIME_UTC, inplace=True)
+        df.index = pd.to_datetime(df.index, utc=True, format='%d.%m.%Y %H:%M ')
+        df.START_TIME_UTC = df.index.to_pydatetime()
+
+        # find column name and choose which to return and which to drop
+        (forecast_load_col, ) = [c for c in df.columns if 'Day-ahead Total Load Forecast [MW]' in c]
+        (actual_load_col, ) = [c for c in df.columns if 'Actual Total Load [MW]' in c]
+        if self.options['forecast']:
+            load_col = forecast_load_col
+            drop_load_col = actual_load_col
+
+            # drop extra columns
+            df = df[self.options['start_at']:self.options['end_at']]
+        else:
+            load_col = actual_load_col
+            drop_load_col = forecast_load_col
+
+        # rename columns for list of dicts return
+        rename_d = {load_col: 'load_MW',
+                    'START_TIME_UTC': 'timestamp',
+                    }
+        df.rename(columns=rename_d, inplace=True)
+        drop_col = ['Time (UTC)', 'END_TIME_UTC', drop_load_col]
+        df.drop(drop_col, axis=1, inplace=True)
+
+        # drop nan rows
+        df.dropna(subset=['load_MW'], inplace=True)
+
+        # Add columns
+        df['ba_name'] = self.options['control_area']
+        df['freq'] = '1hr'
+        df['market'] = 'RTHR'  # not necessarily appropriate terminology
+
+        if self.options['latest']:
+            # drop all but the latest record not in the future
+            now = datetime.now(pytz.utc)
+            df = df[:now].iloc[-1]
+            return [df.to_dict()]
+
+        return df.to_dict('records')
+
+    def get_load(self, control_area='CTA|IT', latest=False, start_at=None, end_at=None,
```

Comment:
I use control area terminology to match entsoe terminology, the closest analog from US terminology is TSO